code/modules/shuttle/shuttle_weapons/weapon_controller.dm

```diff
@@ -0,0 +1,319 @@
+/*
+ * Handles the firing of weapons at hostile ships.
+ * Basically a copy paste of sec camera console, with functionality for weapons, and instead of cameras it looks at tracked docking ports.
+ * Additionally handles declaring ships rogue if they fire upon friendly ships, since its much quicker to see what camera they are on than to find what shuttle a turf is attached to.
+ */
+
+/obj/machinery/computer/weapons
+	name = "weapons control console"
+	desc = "a computer for controlling the weapon systems of your shuttle."
+	icon_screen = "cameras"
+	icon_keyboard = "security_key"
+	circuit = /obj/item/circuitboard/computer/shuttle/weapons
+	light_color = LIGHT_COLOR_RED
+
+	var/list/weapon_weakrefs = list()	//A list of weakrefs to the weapon systems
+	var/shuttle_id						//The shuttle we are connected to
+	var/selected_ship_id = null
+	var/list/concurrent_users = list()
+
+	var/extra_range = 3
+
+	//Weapon systems
+	var/datum/weakref/selected_weapon_system = null
+
+	// Stuff needed to render the map
+	var/map_name
+	var/const/default_map_size = 15
+	//Contents holder to make the turfs clickable :^)
+	var/atom/movable/screen/map_view/weapons_console/cam_screen
+	var/atom/movable/screen/plane_master/lighting/cam_plane_master
+	var/atom/movable/screen/background/cam_background
+
+	//The coords of the top corner
+	var/corner_x
+	var/corner_y
+	var/corner_z
+
+/obj/machinery/computer/weapons/Initialize(mapload, obj/item/circuitboard/C)
+	. = ..()
+	map_name = "weapon_console_[REF(src)]_map"
+	// Initialize map objects
+	cam_screen = new
+	cam_screen.name = "screen"
+	cam_screen.assigned_map = map_name
+	cam_screen.del_on_map_removal = FALSE
+	cam_screen.screen_loc = "[map_name]:1,1"
+	cam_screen.link_to_console(src)
+	cam_plane_master = new
+	cam_plane_master.name = "plane_master"
+	cam_plane_master.assigned_map = map_name
+	cam_plane_master.del_on_map_removal = FALSE
+	cam_plane_master.screen_loc = "[map_name]:CENTER"
+	cam_background = new
+	cam_background.assigned_map = map_name
+	cam_background.del_on_map_removal = FALSE
+	if(!shuttle_id)
+		addtimer(CALLBACK(src, .proc/get_attached_ship), 10)
```

Comment:
Again, this delay might or might not be enough. Make sure there are ways to ensure this happens successfully.